### PR TITLE
k3d migration

### LINKED ISSUE #141: Consider the switch from kind to k3d
We already made some experiments with k3d which looked very good in terms of speed and performance, the only visible blocker was inability to have the test on the same network as in 
https://github.com/rancher/k3d/issues/111

Looks like the issue is solved in recent k3d version and we might give it another shot as we definitely want faster test cycle both locally and within ci/cd pipelines

### PR DESCRIPTION
**Pull request by kuritka**
close  #141 

- upgrade workflows to the newest ubuntu-20.04. For more details, see https://github.com/actions/virtual-environments/issues/1816
- create k3d github action
- remove private registry from makefile
- remove kind configs
- update Makefile (deploy-full-local-setup, deploy-terratest-setup)
- update tests k3d
- update documentation references from kind to k3d, change number of returned nodes in `dig @localhost localtargets-app3.cloud.example.com` from three to four
- upgrade podinfo version
- specify underlying docker bridge network (required by k3d)
- refactor GitHub `terratest.yaml` workflow, currently running against @main until v1.0.0 will be released
- refactor and test `release.yaml` workflow to use k3d action with single cluster, no private docker registry
- passed https://github.com/AbsaOSS/k8gb/blob/master/docs/local.md

### CODE REVIEW

**Comment by ytsarev** [comment]
@kuritka could you please rebase with master. Going to thoroughly test it with recent codebase

**Comment by kuritka** [comment]
> @kuritka could you please rebase with master. Going to thoroughly test it with recent codebase

@ytsarev  rebased

File: Makefile [hunk 1]
```diff
@@ -3,20 +3,17 @@
...
 ETCD_DEBUG_IMAGE ?= quay.io/coreos/etcd:v3.2.25
-CLUSTER_GSLB2_HELM_ARGS ?= --set k8gb.clusterGeoTag='us' --set k8gb.extGslbClustersGeoTags='eu' --set k8gb.hostAlias.hostnames='{gslb-ns-cloud-example-com-eu.example.com}'
-
 
-# terratest
-GITACTION_TERRATEST_DOCKER_REPO_PORT ?= 5000
-GITACTION_TERRATEST_DOCKER_REPO_NAME ?= kind-registry
-GITACTION_TERRATEST_DOCKER_REPO_IMAGE ?= registry:2
+CLUSTER_GSLB2_HELM_ARGS ?= --set k8gb.clusterGeoTag='us' --set k8gb.extGslbClustersGeoTags='eu' --set k8gb.hostAlias.hostnames='{gslb-ns-cloud-example-com-eu.example.com}'
+GITACTION_IMAGE_TAG ?=registry.localhost:5000/k8gb
```
Comment (ytsarev) [inline Makefile:16]: nitpick - it's actually not a tag but a image repo. The tag is commit hash below

File: Makefile [hunk 2]
```diff
@@ -316,7 +311,7 @@ define deploy-local-cluster
 	@echo "\n$(YELLOW)Deploy test apps $(NC)"
 	$(call deploy-test-apps)
 
-	@echo "\n$(YELLOW)Wait for GSLB, Ingress are ready $(NC)"
+	@echo "\n$(YELLOW)Wait for ETCD, Ingress are ready $(NC)"
```
Comment (ytsarev) [inline Makefile:314]:
```suggestion
	@echo "\n$(YELLOW)Wait until ETCD and Ingress controller are ready $(NC)"
```

**Review by ytsarev — CHANGES_REQUESTED** [review]
Some cosmetic suggestions. Amazing job otherwise, works like a charm!

**Review by ytsarev — APPROVED** [review]

**Comment by ytsarev** [comment]
Amazing, so let's merge it and cut-off the release

**Pull request merged**